qutebrowser/browser/network/networkmanager.py

```diff
@@ -358,6 +358,23 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        def same_domain():
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=self._win_id)
+            view = tabbed_browser.currentWidget()
+            print(req.url().host() == view.url().host(), repr(view.url().host()), repr(req.url().host()))
+            # TODO: We probably want to allow headers if we're on test.com and
+            # doing a request to www.test.com? Or maybe a new settings value for
+            # this?
+            return req.url().host() == view.url().host()
```

Comment:
There's [QUrl::topLevelDomain](http://doc.qt.io/qt-5/qurl.html#topLevelDomain) though I wonder how outdated that information is (the data is in [qurltlds_p.h](http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qurltlds_p.h)). The [INFO file](http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qurltlds_p.h.INFO) says it's currently from February 2015, but for Qt 5.2 it might be rather old already...

As for Python libraries, there's [this code on SO](http://stackoverflow.com/a/1069780/2085149) or [tldextract](https://pypi.python.org/pypi/tldextract) or [publicsuffix](https://pypi.python.org/pypi/publicsuffix).

Anyways, I'd say the Qt call should be used for now, and we can always improve it later.